## Summary

### tl;dr

Notes:
* Valkey is a Linux Foundation Project and was forked from Redis on 2024-03-28
* Caveat: This is purely a summary of the data that has not been validated from anyone within the community
* Analysis of contributions to one repo: https://github.com/valkey-io/valkey

Diverse set of contributors from a variety of companies with Amazon and Google employees making the most significant contributions, and no contributions from Redis employees after the fork.

### After 2024-03-28 Fork (until 2024-08-20):

Companies with >= 5 commits: 
* People: 27 0.32926829268292684 % of people employed at 9 companies
* Commits: 327
* Additions: 70708 0.7885885082084225 % of total additions
* Deletions: 50158 0.7620827445796678 % of total deletions

|People|Company|Commits|Additions|Addition%|Deletions|Deletion%|
|:---|:---|:---|:---|:---|:---|:---|
| 2 | Google | 27 | 40191 | 44.82% | 38406 | 58.35% |
| 12 | Amazon | 122 | 15072 | 16.81% | 5772 | 8.77% |
| 2 | Alibaba | 6 | 303 | 0.34% | 51 | 0.08% |
| 2 | ByteDance | 5 | 3913 | 4.36% | 562 | 0.85% |
| 4 | Huawei | 66 | 3205 | 3.57% | 2764 | 4.20% |
| 1 | Tencent Cloud | 46 | 1690 | 1.88% | 703 | 1.07% |
| 1 | Intel | 10 | 559 | 0.62% | 455 | 0.69% |
| 2 | Ericsson | 37 | 5671 | 6.32% | 1372 | 2.08% |
| 1 | @gnet-io | 8 | 104 | 0.12% | 73 | 0.11% |


In [2]:
from pprint import pprint
import collections
import pandas as pd
import pickle

# Pickle files generated by this script:
# https://github.com/geekygirldawn/project-api-metrics/blob/main/scripts/commits_people.py

people_pickle = '../data-files/valkey_people_2024-03-28T00:00:00.000+00:002024-08-20T00:00:00.000+00:00.pkl'

with open(people_pickle, 'rb') as f:
    person_dict = pickle.load(f)
    
len(person_dict)

82

In [3]:
people = len(person_dict)
commits = 0
additions = 0
deletions = 0

for key,value in person_dict.items():
    # Normalize company names and use emails to derive some affiliations
    print(key,value)
    if value['company'] == None:
        for email in value['email']:
            if any(x in email.lower() for x in ['redis.com', 'redislabs.com']):
                person_dict[key]['company'] = 'Redis Labs'
            if 'amazon.com' in email:
                person_dict[key]['company'] = 'Amazon'
    elif 'huawei' in value['company'].lower():
        person_dict[key]['company'] = 'Huawei'
    elif 'alibaba' in value['company'].lower():
        person_dict[key]['company'] = 'Alibaba'
    elif any(x in value['company'].lower() for x in ['aws','amazon']):
        person_dict[key]['company'] = 'Amazon'
    elif 'ericsson' in value['company'].lower():
        person_dict[key]['company'] = 'Ericsson'
    elif 'samsung' in value['company'].lower():
        person_dict[key]['company'] = 'Samsung'
        
    # Get descriptive statistics
    commits = commits + value['commits']
    additions = additions + value['additions']
    deletions = deletions + value['deletions']
    
print("People:", people)
print("Commits:", commits)
print("Additions", additions)
print("Deletions", deletions)

shogo82148 {'email': ['shogo82148@gmail.com'], 'name': 'ICHINOSE Shogo', 'company': '@fuller-inc ', 'additions': 41, 'deletions': 38, 'commits': 1}
PingXie {'email': ['pingxie@outlook.com', 'pingxie@google.com'], 'name': 'Ping Xie', 'company': 'Google', 'additions': 39434, 'deletions': 37649, 'commits': 26}
madolson {'email': ['madelyneolson@gmail.com', '34459052+madolson@users.noreply.github.com'], 'name': 'Madelyn Olson', 'company': 'Amazon Web Services', 'additions': 8308, 'deletions': 3604, 'commits': 65}
pieturin {'email': ['105686771+pieturin@users.noreply.github.com'], 'name': 'Pierre', 'company': 'Amazon Web Services', 'additions': 5, 'deletions': 0, 'commits': 1}
bentotten {'email': ['59932872+bentotten@users.noreply.github.com'], 'name': None, 'company': None, 'additions': 5, 'deletions': 4, 'commits': 4}
eliblight {'email': ['eran.liberty@gmail.com'], 'name': 'Eran Liberty', 'company': None, 'additions': 14, 'deletions': 48, 'commits': 5}
stockholmux {'email': ['kyledvs@amaz

In [4]:
for key,value in person_dict.items():
    if (value['commits'] >= 5) and (value['company'] == None):
        print(key,value)
        
# Note: Unaffiliated after Aug 2023: https://www.linkedin.com/in/9bany/

eliblight {'email': ['eran.liberty@gmail.com'], 'name': 'Eran Liberty', 'company': None, 'additions': 14, 'deletions': 48, 'commits': 5}
uriyage {'email': ['78144248+uriyage@users.noreply.github.com'], 'name': None, 'company': None, 'additions': 2053, 'deletions': 1044, 'commits': 7}
9bany {'email': ['53461381+0del@users.noreply.github.com'], 'name': 'Bany', 'company': None, 'additions': 619, 'deletions': 616, 'commits': 26}
vitahlin {'email': ['vitahlin@gmail.com', 'linw1225@gmail.com'], 'name': 'Vitah Lin', 'company': None, 'additions': 144, 'deletions': 57, 'commits': 9}
roshkhatri {'email': ['117414976+roshkhatri@users.noreply.github.com'], 'name': 'Roshan Khatri', 'company': None, 'additions': 767, 'deletions': 113, 'commits': 7}


In [5]:
# Manual Fixes
person_dict['roshkhatri']['company'] = 'Amazon' # https://www.linkedin.com/in/roshan-vivek-khatri-a403a8193/
person_dict['uriyage']['company'] = 'Amazon' # https://valkey.io/blog/unlock-one-million-rps/
person_dict['eliblight']['company'] = 'Amazon' # https://www.linkedin.com/in/eran-liberty-b7317b4/
person_dict['stockholmux']['company'] = 'Amazon' # Using amazon.com email

In [6]:
import collections

org_people = 0
org_commits = 0
org_additions = 0
org_deletions = 0

orgs = 0

org_dict = collections.defaultdict(dict)

for key,value in person_dict.items():
    try:
        org_dict[value['company']]['commits'] = org_dict[value['company']]['commits'] + value['commits']
        org_dict[value['company']]['additions'] = org_dict[value['company']]['additions'] + value['additions']
        org_dict[value['company']]['deletions'] = org_dict[value['company']]['deletions'] + value['deletions']
        org_dict[value['company']]['people'].append(key)
    except:
        org_dict[value['company']]['commits'] = value['commits']
        org_dict[value['company']]['additions'] = value['additions']
        org_dict[value['company']]['deletions'] = value['deletions']
        org_dict[value['company']]['people'] = [key]

        
# Make it easy for the print statements to be copied into a Markdown table
print('|People|Company|Commits|Additions|Addition%|Deletions|Deletion%|')
print('|:---|:---|:---|:---|:---|:---|:---|')

for key,value in org_dict.items():
    if value['commits'] >= 5:
        if key != None:
            org_people = org_people + len(value['people'])
            org_commits = org_commits + value['commits']
            org_additions = org_additions + value['additions']
            org_deletions = org_deletions + value['deletions']
            orgs+=1
            add_perc = format(value['additions']/additions, ".2%")
            del_perc = format(value['deletions']/deletions, ".2%")
            print('|', len(value['people']),'|', key, '|', value['commits'],'|', value['additions'],'|',add_perc,'|', value['deletions'],'|',del_perc,'|')

print("\nCompanies with >= 5 commits:", "\n* People:", org_people, org_people/people, "% of people employed at", orgs, "companies")
print("* Commits:", org_commits)
print("* Additions:", org_additions, org_additions/additions, "% of total additions")
print("* Deletions:", org_deletions, org_deletions/deletions, "% of total deletions")

|People|Company|Commits|Additions|Addition%|Deletions|Deletion%|
|:---|:---|:---|:---|:---|:---|:---|
| 2 | Google | 27 | 40191 | 44.82% | 38406 | 58.35% |
| 12 | Amazon | 122 | 15072 | 16.81% | 5772 | 8.77% |
| 2 | Alibaba | 6 | 303 | 0.34% | 51 | 0.08% |
| 2 | ByteDance | 5 | 3913 | 4.36% | 562 | 0.85% |
| 4 | Huawei | 66 | 3205 | 3.57% | 2764 | 4.20% |
| 1 | Tencent Cloud | 46 | 1690 | 1.88% | 703 | 1.07% |
| 1 | Intel | 10 | 559 | 0.62% | 455 | 0.69% |
| 2 | Ericsson | 37 | 5671 | 6.32% | 1372 | 2.08% |
| 1 | @gnet-io | 8 | 104 | 0.12% | 73 | 0.11% |

Companies with >= 5 commits: 
* People: 27 0.32926829268292684 % of people employed at 9 companies
* Commits: 327
* Additions: 70708 0.7885885082084225 % of total additions
* Deletions: 50158 0.7620827445796678 % of total deletions


In [11]:
i = 0
# Make it easy for the print statements to be copied into a Markdown table
print('|People|Company|Commits|Additions|Addition%|Deletions|Deletion%|')
print('|:---|:---|:---|:---|:---|:---|:---|')

for key,value in org_dict.items():
           
    add_perc = format(value['additions']/additions, ".2%")
    del_perc = format(value['deletions']/deletions, ".2%")
    print('|', len(value['people']),'|', key, '|', value['commits'],'|', value['additions'],'|',add_perc,'|', value['deletions'],'|',del_perc,'|')

    i+=1

print("Number of companies:", i)

|People|Company|Commits|Additions|Addition%|Deletions|Deletion%|
|:---|:---|:---|:---|:---|:---|:---|
| 1 | @fuller-inc  | 1 | 41 | 0.05% | 38 | 0.06% |
| 2 | Google | 27 | 40191 | 44.82% | 38406 | 58.35% |
| 12 | Amazon | 122 | 15072 | 16.81% | 5772 | 8.77% |
| 34 | None | 89 | 4520 | 5.04% | 1592 | 2.42% |
| 2 | Alibaba | 6 | 303 | 0.34% | 51 | 0.08% |
| 1 | Redis | 1 | 175 | 0.20% | 185 | 0.28% |
| 1 | @franklin-ai | 4 | 137 | 0.15% | 96 | 0.15% |
| 2 | ByteDance | 5 | 3913 | 4.36% | 562 | 0.85% |
| 1 | Shanghai Jiaotong University | 1 | 2 | 0.00% | 0 | 0.00% |
| 4 | Huawei | 66 | 3205 | 3.57% | 2764 | 4.20% |
| 1 | LINE | 3 | 122 | 0.14% | 69 | 0.10% |
| 1 | Soochow University | 1 | 13 | 0.01% | 6 | 0.01% |
| 1 | Prunny Technologies | 3 | 1593 | 1.78% | 1585 | 2.41% |
| 1 | @VelocityLimitless | 1 | 5 | 0.01% | 0 | 0.00% |
| 1 | Oracle | 4 | 10948 | 12.21% | 10948 | 16.63% |
| 1 | Tencent Cloud | 46 | 1690 | 1.88% | 703 | 1.07% |
| 1 | Looking for a job | 4 | 170 | 0.19% | 22 | 0.03

## All Organizational Affiliations

Note: In some cases, these are based on what people have put in their GitHub profile and aren't actual organizations (e.g., "Looking for a job"), but those are part of the long tail and aren't significant contributors.

|People|Company|Commits|Additions|Addition%|Deletions|Deletion%|
|:---|:---|:---|:---|:---|:---|:---|
| 1 | @fuller-inc  | 1 | 41 | 0.05% | 38 | 0.06% |
| 2 | Google | 27 | 40191 | 44.82% | 38406 | 58.35% |
| 12 | Amazon | 122 | 15072 | 16.81% | 5772 | 8.77% |
| 34 | None | 89 | 4520 | 5.04% | 1592 | 2.42% |
| 2 | Alibaba | 6 | 303 | 0.34% | 51 | 0.08% |
| 1 | Redis | 1 | 175 | 0.20% | 185 | 0.28% |
| 1 | @franklin-ai | 4 | 137 | 0.15% | 96 | 0.15% |
| 2 | ByteDance | 5 | 3913 | 4.36% | 562 | 0.85% |
| 1 | Shanghai Jiaotong University | 1 | 2 | 0.00% | 0 | 0.00% |
| 4 | Huawei | 66 | 3205 | 3.57% | 2764 | 4.20% |
| 1 | LINE | 3 | 122 | 0.14% | 69 | 0.10% |
| 1 | Soochow University | 1 | 13 | 0.01% | 6 | 0.01% |
| 1 | Prunny Technologies | 3 | 1593 | 1.78% | 1585 | 2.41% |
| 1 | @VelocityLimitless | 1 | 5 | 0.01% | 0 | 0.00% |
| 1 | Oracle | 4 | 10948 | 12.21% | 10948 | 16.63% |
| 1 | Tencent Cloud | 46 | 1690 | 1.88% | 703 | 1.07% |
| 1 | Looking for a job | 4 | 170 | 0.19% | 22 | 0.03% |
| 1 | Humana | 1 | 1087 | 1.21% | 984 | 1.50% |
| 1 | The Linux Foundation | 1 | 6 | 0.01% | 0 | 0.00% |
| 1 | @bandism  | 1 | 1 | 0.00% | 1 | 0.00% |
| 1 | Intel | 10 | 559 | 0.62% | 455 | 0.69% |
| 1 | University of Michigan | 1 | 4 | 0.00% | 8 | 0.01% |
| 1 | @aiven | 1 | 1 | 0.00% | 4 | 0.01% |
| 2 | Ericsson | 37 | 5671 | 6.32% | 1372 | 2.08% |
| 2 | Samsung | 3 | 40 | 0.04% | 45 | 0.07% |
| 1 | simkus.io | 2 | 38 | 0.04% | 38 | 0.06% |
| 1 | iandouglas.com | 1 | 2 | 0.00% | 2 | 0.00% |
| 1 | @gnet-io | 8 | 104 | 0.12% | 73 | 0.11% |
| 1 | @kunai-consulting | 1 | 15 | 0.02% | 0 | 0.00% |
| 1 | Seeking Opportunities | 3 | 36 | 0.04% | 36 | 0.05% |